--- a/include/configs/ap121.h
+++ b/include/configs/ap121.h
@@ -358,7 +358,14 @@
 /* For Kite, only PCI-e interface is valid */
 #define AR7240_ART_PCICFG_OFFSET        3
 
-
+#define CONFIG_BOOTDELAY	    	3	
+#define CONFIG_BOOT_RETRY_TIME		-1
+#define CONFIG_AUTOBOOT_KEYED
+#define CONFIG_AUTOBOOT_PROMPT		"autoboot in %d seconds (stop with 'lin')...\n"
+#define CONFIG_AUTOBOOT_STOP_STR	"lin"
+#define CONFIG_ZERO_BOOTDELAY_CHECK
+#define CONFIG_RESET_TO_RETRY		1
+#define CONFIG_BOOT_RETRY_MIN		1
 
 #include <cmd_confdefs.h>
 
