Index: u-boot-linino/include/configs/ap121.h
===================================================================
--- u-boot-linino.orig/include/configs/ap121.h	2014-06-04 18:13:40.000000000 +0200
+++ u-boot-linino/include/configs/ap121.h	2014-06-04 18:14:30.000000000 +0200
@@ -79,6 +79,7 @@
 #define	CFG_MONITOR_BASE	TEXT_BASE
 #define	CFG_MONITOR_LEN		(192 << 10)
 
+#ifndef LININO_BOARDS
 #undef CONFIG_BOOTARGS
 /* XXX - putting rootfs in last partition results in jffs errors */
 
@@ -105,6 +106,7 @@
 #define	CONFIG_BOOTARGS     "console=ttyS0,115200 root=31:01 rootfstype=squashfs init=/sbin/init mtdparts=ar7240-nor0:64k(u-boot),3008k(rootfs),896k(uImage),64k(mib0),64k(ART)"
 #define MTDPARTS_DEFAULT    "mtdparts=ar7240-nor0:64k(u-boot),3008k(rootfs),896k(uImage),64k(NVRAM),64k(ART)"
 #endif
+#endif //LININO_BOARDS
 
 #undef CFG_PLL_FREQ
 
