Index: u-boot-linino/Makefile
===================================================================
--- u-boot-linino.orig/Makefile	2014-06-04 18:26:24.000000000 +0200
+++ u-boot-linino/Makefile	2014-06-04 18:26:24.000000000 +0200
@@ -341,6 +341,7 @@
 ifdef BOARD_STRING
 	@echo "#define BOARD_STRING 1" >>include/config.h
 endif
+	@echo "#define LININO_BOARDS 1" >>include/config.h
 	@./mkconfig -a ap121 mips mips ap121 ar7240 ar7240 linino
 	
 #########################################################################
Index: u-boot-linino/include/configs/ap121.h
===================================================================
--- u-boot-linino.orig/include/configs/ap121.h	2014-06-04 18:26:36.000000000 +0200
+++ u-boot-linino/include/configs/ap121.h	2014-06-04 18:30:37.000000000 +0200
@@ -79,13 +79,13 @@
 #define	CFG_MONITOR_BASE	TEXT_BASE
 #define	CFG_MONITOR_LEN		(192 << 10)
 
-#ifndef LININO_BOARDS
 #undef CONFIG_BOOTARGS
 /* XXX - putting rootfs in last partition results in jffs errors */
 
 /* default mtd partition table */
 #undef MTDPARTS_DEFAULT
 
+#ifndef LININO_BOARDS
 #if (FLASH_SIZE == 16)
 #define CONFIG_BOOTARGS     "console=ttyS0,115200 root=31:02 rootfstype=squashfs init=/sbin/init mtdparts=ar7240-nor0:256k(u-boot),64k(u-boot-env),14656k(rootfs),1280k(uImage),64k(NVRAM),64k(ART)"
 #define MTDPARTS_DEFAULT    "mtdparts=ar7240-nor0:256k(u-boot),64k(u-boot-env),14656k(rootfs),1280k(uImage),64k(NVRAM),64k(ART)"
@@ -106,6 +106,12 @@
 #define	CONFIG_BOOTARGS     "console=ttyS0,115200 root=31:01 rootfstype=squashfs init=/sbin/init mtdparts=ar7240-nor0:64k(u-boot),3008k(rootfs),896k(uImage),64k(mib0),64k(ART)"
 #define MTDPARTS_DEFAULT    "mtdparts=ar7240-nor0:64k(u-boot),3008k(rootfs),896k(uImage),64k(NVRAM),64k(ART)"
 #endif
+
+#else
+
+#define CONFIG_BOOTARGS     "console=spicons board=linino-yun mem=64M rootfstype=squashfs,jffs2 noinitrd mtdparts=spi0.0:256k(u-boot)ro,64k(u-boot-env),14656k(rootfs),1280k(kernel),64k(nvram),64k(art),15936k@0x50000(firmware)"
+#define MTDPARTS_DEFAULT    "mtdparts=spi0.0:256k(u-boot)ro,64k(u-boot-env)ro,14656k(rootfs),1280k(kernel),64k(nvram),64k(art),15936k@0x50000(firmware)"
+
 #endif //LININO_BOARDS
 
 #undef CFG_PLL_FREQ
