From ddc23b81bb7ce5e1c16326aa5509dd92b7fee4a4 Mon Sep 17 00:00:00 2001
From: Federico Vaga <federico.vaga@gmail.com>
Date: Tue, 3 Dec 2013 21:21:36 +0100
Subject: [PATCH] error_module.cpp: remove useless include

Signed-off-by: Federico Vaga <federico.vaga@gmail.com>
---
 error_module.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/error_module.cpp b/error_module.cpp
index 6c40900..eb26793 100644
--- a/error_module.cpp
+++ b/error_module.cpp
@@ -28,7 +28,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <signal.h>
-#include <execinfo.h>
 
 #include "libs/error/error.h"
 #include "libs/sys/module.h"
-- 
1.8.3.1

