--- a/arch/mips/ath79/Kconfig
+++ b/arch/mips/ath79/Kconfig
@@ -648,6 +648,71 @@ config ATH79_MACH_NBG460N
 	select ATH79_DEV_M25P80
 	select ATH79_DEV_WMAC
 
+config ATH79_MACH_LININO_YUN
+	bool "Arduino YUN support"
+	select SOC_AR933X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+	help
+		Say 'Y' here if you want your kernel to support
+		Arduino Yun.
+
+config ATH79_MACH_LININO_ONE
+	bool "Linino One support"
+	select SOC_AR933X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+	help
+		Say 'Y' here if you want your kernel to support
+		Linino One.
+
+config ATH79_MACH_LININO_FREEDOG
+	bool "Linino Freedog support"
+	select SOC_AR933X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+	help
+		Say 'Y' here if you want your kernel to support
+	 	Linino Freedog.
+
+config ATH79_MACH_LININO_CHOWCHOW
+	bool "Linino Chowchow support"
+	select SOC_AR934X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+	help
+		Say 'Y' here if you want your kernel to support
+		Linino Chowchow.
+
+config ATH79_MACH_LININO_CHIWAWA
+	bool "Linino Chiwawa support"
+	select SOC_AR933X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+	help
+		Say 'Y' here if you want your kernel to support
+		Linino Chiwawa.
+
 endmenu
 
 config SOC_AR71XX
--- a/arch/mips/ath79/machtypes.h
+++ b/arch/mips/ath79/machtypes.h
@@ -117,6 +117,11 @@ enum ath79_mach_type {
 	ATH79_MACH_WZR_HP_G450H,	/* Buffalo WZR-HP-G450H */
 	ATH79_MACH_ZCN_1523H_2,		/* Zcomax ZCN-1523H-2-xx */
 	ATH79_MACH_ZCN_1523H_5,		/* Zcomax ZCN-1523H-5-xx */
+	ATH79_MACH_LININO_YUN,		/* Linino Yun */
+	ATH79_MACH_LININO_ONE,		/* Linino One */
+	ATH79_MACH_LININO_FREEDOG,	/* Linino Freedog */
+	ATH79_MACH_LININO_CHOWCHOW,	/* Linino Chowchow */
+	ATH79_MACH_LININO_CHIWAWA,	/* Linino Chiwawa */
 };
 
 #endif /* _ATH79_MACHTYPE_H */
--- a/arch/mips/ath79/Makefile
+++ b/arch/mips/ath79/Makefile
@@ -59,6 +59,11 @@ obj-$(CONFIG_ATH79_MACH_EAP7660D)	+= mac
 obj-$(CONFIG_ATH79_MACH_JA76PF)		+= mach-ja76pf.o
 obj-$(CONFIG_ATH79_MACH_JWAP003)	+= mach-jwap003.o
 obj-$(CONFIG_ATH79_MACH_HORNET_UB)	+= mach-hornet-ub.o
+obj-$(CONFIG_ATH79_MACH_LININO_YUN)	+= mach-linino-yun.o
+obj-$(CONFIG_ATH79_MACH_LININO_ONE)	+= mach-linino-yun.o
+obj-$(CONFIG_ATH79_MACH_LININO_FREEDOG)	+= mach-linino-freedog.o
+obj-$(CONFIG_ATH79_MACH_LININO_CHOWCHOW)+= mach-linino-chowchow.o
+obj-$(CONFIG_ATH79_MACH_LININO_CHIWAWA) += mach-linino-chiwawa.o
 obj-$(CONFIG_ATH79_MACH_MR600)		+= mach-mr600.o
 obj-$(CONFIG_ATH79_MACH_MZK_W04NU)	+= mach-mzk-w04nu.o
 obj-$(CONFIG_ATH79_MACH_MZK_W300NH)	+= mach-mzk-w300nh.o
