From e05de99d21b5fd99bd3b0c064fd0e962e3e0fd1f Mon Sep 17 00:00:00 2001
From: Aurelio Colosimo <aurelio@aureliocolosimo.it>
Date: Mon, 30 Jun 2014 10:13:55 +0200
Subject: [PATCH] drivers/spi/spi-tty-plain.c: destroy spi_pollwq at remove

Signed-off-by: Aurelio Colosimo <aurelio@aureliocolosimo.it>
---
 drivers/spi/spi-tty-plain.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-tty-plain.c b/drivers/spi/spi-tty-plain.c
index 29db177..4902811 100644
--- a/drivers/spi/spi-tty-plain.c
+++ b/drivers/spi/spi-tty-plain.c
@@ -299,8 +299,7 @@ static int spi_tty_probe(struct spi_device *spi)
 	pr_info("spi_poll_wq called every %u jiffies\n",
 		(unsigned)wqinterval);
 
-	if (!stty->wq)
-		stty->wq = create_singlethread_workqueue("spi_poll_wq");
+	stty->wq = create_singlethread_workqueue("spi_poll_wq");
 
 	if (stty->wq) {
 		INIT_DELAYED_WORK(&stty->work, spi_poll_work_handler);
@@ -321,6 +320,7 @@ static int spi_tty_remove(struct spi_device *spi)
 
 	cancel_delayed_work_sync(&stty->work);
 	flush_workqueue(stty->wq);
+	destroy_workqueue(stty->wq);
 
 	dev_info(&spi->dev, "%s\n", __func__);
 
-- 
1.7.9.5

