From 080acff68b9a516b8bc1694bf93d2c4dd77c972f Mon Sep 17 00:00:00 2001
From: Aurelio Colosimo <aurelio@aureliocolosimo.it>
Date: Mon, 30 Jun 2014 16:19:43 +0200
Subject: [PATCH 1/2] drivers/mcuio/mcuio-adc.c: declare all attr functions as
 static

Signed-off-by: Aurelio Colosimo <aurelio@aureliocolosimo.it>
---
 drivers/mcuio/mcuio-adc.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/mcuio/mcuio-adc.c b/drivers/mcuio/mcuio-adc.c
index 746dc7b..b776e97 100644
--- a/drivers/mcuio/mcuio-adc.c
+++ b/drivers/mcuio/mcuio-adc.c
@@ -145,7 +145,7 @@ struct attribute_group adc_in_dev_attr_group = {
 	.attrs = adc_in_dev_attrs,
 };
 
-ssize_t show_enable(struct device *dev, struct device_attribute *attr,
+static ssize_t show_enable(struct device *dev, struct device_attribute *attr,
 			char *buf)
 {
 	uint32_t en = 0;
@@ -203,7 +203,7 @@ static void in_devs_destroy(struct adc_data *data)
 	mcuio_adc_class = NULL;
 }
 
-ssize_t store_enable(struct device *dev, struct device_attribute *attr,
+static ssize_t store_enable(struct device *dev, struct device_attribute *attr,
 			 const char *buf, size_t count)
 {
 	uint32_t _en, en_new;
@@ -229,14 +229,14 @@ ssize_t store_enable(struct device *dev, struct device_attribute *attr,
 	return count;
 }
 
-ssize_t show_aref(struct device *dev, struct device_attribute *attr,
+static ssize_t show_aref(struct device *dev, struct device_attribute *attr,
 			char *buf)
 {
 	/* FIXME todo */
 	return sprintf(buf, "Not implemented yet");
 }
 
-ssize_t store_aref(struct device *dev, struct device_attribute *attr,
+static ssize_t store_aref(struct device *dev, struct device_attribute *attr,
 			 const char *buf, size_t count)
 {
 	/* FIXME todo */
-- 
1.7.9.5

