From bd02bdbaa71e34c128007f2979db9c97f70de0b0 Mon Sep 17 00:00:00 2001
From: Aurelio Colosimo <aurelio@aureliocolosimo.it>
Date: Thu, 31 Jul 2014 15:11:58 +0200
Subject: [PATCH 50/51] MCUIO pwm: implement set_polarity function

Signed-off-by: Aurelio Colosimo <aurelio@aureliocolosimo.it>
---
 drivers/pwm/pwm-mcuio.c |   20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/drivers/pwm/pwm-mcuio.c b/drivers/pwm/pwm-mcuio.c
index 3a6e017..14ec2fd 100644
--- a/drivers/pwm/pwm-mcuio.c
+++ b/drivers/pwm/pwm-mcuio.c
@@ -79,7 +79,25 @@ done:
 static int mcuio_pwm_set_polarity(struct pwm_chip *chip, struct pwm_device *pwm,
 				  enum pwm_polarity polarity)
 {
-	return -EINVAL;
+	struct mcuio_pwm_data *data = to_mcuio_pwm_data(chip);
+	int idx = pwm_idx(pwm);
+	u32 st;
+	u32 addr;
+
+	addr = 0x040 * (idx + 1) + 0x0c;
+
+	if (regmap_read(data->map, addr, &st))
+		return -EIO;
+
+	if (polarity == PWM_POLARITY_NORMAL)
+		st &= ~(1 << 1);
+	else
+		st |= (1 << 1);
+
+	if (regmap_write(data->map, addr, st))
+		return -EIO;
+
+	return 0;
 }
 
 static int mcuio_pwm_enable(struct pwm_chip *chip, struct pwm_device *pwm)
-- 
1.7.9.5

