From 936b2b1ed1f4f959ae1b2796073a4b9181d45f5a Mon Sep 17 00:00:00 2001
From: Davide Ciminaghi <ciminaghi@gnudd.com>
Date: Thu, 23 Oct 2014 18:41:57 +0200
Subject: [PATCH 172/188] MCUIO, Kconfig: fix Kconfig for menu rendering

---
 drivers/mcuio/Kconfig |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/mcuio/Kconfig b/drivers/mcuio/Kconfig
index 70c3883..98cbc31 100644
--- a/drivers/mcuio/Kconfig
+++ b/drivers/mcuio/Kconfig
@@ -7,9 +7,10 @@ menuconfig MCUIO
 	  (MCU = Micro Controller Unit, some kind of tiny microcontroller
 	  running a tiny OS or no OS at all).
 
+if MCUIO
+
 config MCUIO_LDISC_HC
 	tristate "mcuio line discipline host controller"
-	depends on MCUIO
 	default y
 	help
 	  This adds a line-discipline based MCUIO host controller, which is
@@ -17,21 +18,20 @@ config MCUIO_LDISC_HC
 
 config MCUIO_JS_SHIELD
 	tristate "mcuio joystick shield driver"
-	depends on (MCUIO && MCUIO_IRQ_CTRL)
 	default y
 	help
 	  Adds a driver for a joystick + buttons shield (experimental)
 
 config MCUIO_LUCKY_SHIELD
 	tristate "mcuio lucky shield driver"
-	depends on (MCUIO && MCUIO_IRQ_CTRL && I2C)
+	depends on I2C
 	default y
 	help
 	  Adds a driver for the Dog Hunter Lucky shield (experimental)
 
 config MCUIO_DIGITALIO_SHIELD
 	tristate "mcuio Digital I/O shield driver"
-	depends on (MCUIO && MCUIO_IRQ_CTRL && I2C)
+	depends on I2C
 	default y
 	help
 	  Adds a driver for the Dog Hunter Digital IO shield (experimental)
@@ -53,7 +53,8 @@ config MCUIO_SHIELDS_MANUAL_PROBE
 
 config MCUIO_IRQ_TEST
         tristate "mcuio irq test function driver"
-	depends on MCUIO
 	default n
 	help
 	  Adds a driver for the irq test mcuio function
+
+endif # MCUIO
-- 
1.7.9.5

