From 5affae07055260bad0f05fd9de4a49d9bec9090a Mon Sep 17 00:00:00 2001
From: Aurelio Colosimo <aurelio@aureliocolosimo.it>
Date: Tue, 1 Dec 2015 12:11:28 +0100
Subject: [PATCH 42/50] TTY on SPI driver: set std termios settings at tty init

Signed-off-by: Aurelio Colosimo <aurelio@aureliocolosimo.it>
---
 drivers/spi/spi-tty-plain.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-tty-plain.c b/drivers/spi/spi-tty-plain.c
index 4c60724..3a3e653 100644
--- a/drivers/spi/spi-tty-plain.c
+++ b/drivers/spi/spi-tty-plain.c
@@ -385,6 +385,7 @@ static int spi_tty_init(void)
 	spi_serial_tty_driver->type = TTY_DRIVER_TYPE_SERIAL;
 	spi_serial_tty_driver->subtype = SERIAL_TYPE_NORMAL;
 	spi_serial_tty_driver->flags = TTY_DRIVER_DYNAMIC_DEV;
+	spi_serial_tty_driver->init_termios = tty_std_termios;
 
 	tty_set_operations(spi_serial_tty_driver, &spi_serial_ops);
 	err = tty_register_driver(spi_serial_tty_driver);
-- 
2.8.2

